homeassistant/components/device_tracker/__init__.py

```diff
@@ -382,7 +382,7 @@ def async_update_stale(self, now: dt_util.dt.datetime):
         This method must be run in the event loop.
         """
         for device in self.devices.values():
-            if (device.track and device.last_update_home) and \
+            if (device.track and (device.state == STATE_HOME)) and \
```

Comment:
This won't work because we're currently conflating the concerns of location based trackers and network based trackers, meaning a location set to HOME by a location tracker should never be overriden to be marked as stale or last seen updated.

We have decoupled the two implementations so that future platforms will not have this issue. Older platforms will be slowly migrated to the new style.